std/algorithm/iteration.d

```diff
@@ -4559,10 +4559,10 @@ if (isInputRange!R && !isInfinite!R)
     {
         foreach (k; 0 .. data.length / 16)
         {
-            static if (isRandomAccessRange!R && hasSlicing!R)
+            static if (isRandomAccessRange!R && hasLength!R && hasSlicing!R)
             {
                 store[idx] = sumPairwise16!F(data);
-                data = data[16 .. $];
```

Comment:
Isn't that a language bug? $ in slices should be defined by the compiler for all arrays - who is going to need to overwrite this?